### PR TITLE
Add new `Capybara/RSpec/PredicateMatcher` cop

### PR DESCRIPTION
**Pull request by ydah**
Re-targeting: https://github.com/rubocop/rubocop-rspec/pull/1495

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `main` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

- [x] Added the new cop to `config/default.yml`.
- [x] The cop is configured as `Enabled: pending` in `config/default.yml`.
- [x] The cop documents examples of good and bad code.
- [x] The tests assert both that bad code is reported and that good code is not reported.
- [x] Set `VersionAdded: "<<next>>"` in `default/config.yml`.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ task :build_config do
   require 'rubocop/capybara/config_formatter'
   require 'rubocop/capybara/description_extractor'
 
-  glob = File.join('lib', 'rubocop', 'cop', 'capybara', '*.rb')
+  glob = File.join('lib', 'rubocop', 'cop', 'capybara', '{,rspec}', '*.rb')
```
Comment (pirj) [inline Rakefile:36]: 👍 
I guess only SpecificFinders/SpecificActions are not specific to RSpec.

File: lib/rubocop/cop/capybara/rspec/predicate_matcher.rb [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+module RuboCop
+  module Cop
+    module Capybara
+      module RSpec
+        # A helper for `inflected` style
+        module InflectedHelper
+          extend NodePattern::Macros
+
+          EXPLICIT_MATCHER = %w[css selector style xpath].each.map do |suffix|
```
Comment (pirj) [inline lib/rubocop/cop/capybara/rspec/predicate_matcher.rb:11]:
```suggestion
          EXPLICIT_MATCHER = %w[css selector style xpath].map do |suffix|
```

File: lib/rubocop/cop/capybara/rspec/predicate_matcher.rb [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+          def boolean_matcher?(node)
+            if cop_config['Strict']
+              be_boolthy?(node)
+            else
+              be_bool?(node) || be_boolthy?(node)
+            end
+          end
+
+          def predicate?(sym)
+            EXPLICIT_MATCHER.include?(sym.to_s)
```
Comment (pirj) [inline lib/rubocop/cop/capybara/rspec/predicate_matcher.rb:64]: Maybe pre-convert those to `to_sym` and skip `to_s` here?

**Review by pirj — APPROVED** [review]
Thank you!
All notes are optional.
Please merge at will.

File: lib/rubocop/cop/capybara/rspec/predicate_matcher.rb (hunk 3 above)
Comment (ydah) [reply lib/rubocop/cop/capybara/rspec/predicate_matcher.rb]: Indeed it is, I updated this PR 👍🏻

File: lib/rubocop/cop/capybara/rspec/predicate_matcher.rb (hunk 2 above)
Comment (ydah) [reply lib/rubocop/cop/capybara/rspec/predicate_matcher.rb]: Thank you! I updated this PR.

**Comment by pirj** [comment]
Thank you!

**Pull request merged**